tests/lib/rules/max-statements.js

```diff
@@ -34,20 +34,20 @@ ruleTester.run("max-statements", rule, {
-        { code: "var foo = () => { var bar = 1; var baz = 2; var qux = 3; };", options: [2], parserOptions: { ecmaVersion: 6 }, errors: [{ message: "This function has too many statements (3). Maximum allowed is 2."}] },
-        { code: "var foo = function() { var bar = 1; var baz = 2; var qux = 3; };", options: [2], errors: [{ message: "This function has too many statements (3). Maximum allowed is 2."}] },
-        { code: "function foo() { var bar = 1; if (true) { while (false) { var qux = null; } } return 3; }", options: [4], errors: [{ message: "This function has too many statements (5). Maximum allowed is 4."}] },
-        { code: "function foo() { var bar = 1; if (true) { for (;;) { var qux = null; } } return 3; }", options: [4], errors: [{ message: "This function has too many statements (5). Maximum allowed is 4."}] },
-        { code: "function foo() { var bar = 1; if (true) { for (;;) { var qux = null; } } else { quxx(); } return 3; }", options: [5], errors: [{ message: "This function has too many statements (6). Maximum allowed is 5."}] },
-        { code: "function foo() { var x = 5; function bar() { var y = 6; } bar(); z = 10; baz(); }", options: [3], errors: [{ message: "This function has too many statements (5). Maximum allowed is 3."}] },
-        { code: "function foo() { var x = 5; function bar() { var y = 6; } bar(); z = 10; baz(); }", options: [4], errors: [{ message: "This function has too many statements (5). Maximum allowed is 4."}] },
-        { code: ";(function() { var bar = 1; return function () { var z; return 42; }; })()", options: [1, {ignoreTopLevelFunctions: true}], errors: [{ message: "This function has too many statements (2). Maximum allowed is 1."}] },
-        { code: ";(function() { var bar = 1; var baz = 2; })(); (function() { var bar = 1; var baz = 2; })()", options: [1, {ignoreTopLevelFunctions: true}], errors: [{ message: "This function has too many statements (2). Maximum allowed is 1."}, { message: "This function has too many statements (2). Maximum allowed is 1."}] },
-        { code: "define(['foo', 'qux'], function(foo, qux) { var bar = 1; var baz = 2; return function () { var z; return 42; }; })", options: [1, {ignoreTopLevelFunctions: true}], errors: [{ message: "This function has too many statements (2). Maximum allowed is 1."}] },
-        { code: "function foo() { var a; var b; var c; var x; var y; var z; bar(); baz(); qux(); quxx(); foo(); }", errors: [{ message: "This function has too many statements (11). Maximum allowed is 10."}] },
+        { code: "function foo() { var bar = 1; var baz = 2; var qux = 3; }", options: [2], errors: [{ message: "Function 'foo' has too many statements (3). Maximum allowed is 2."}] },
+        { code: "var foo = () => { var bar = 1; var baz = 2; var qux = 3; };", options: [2], parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Function 'anonymous' has too many statements (3). Maximum allowed is 2."}] },
```

Comment:
Based on the comments on #7260, it looks like the decision was to use "this function" in the case of anonymous functions. Can you update that?